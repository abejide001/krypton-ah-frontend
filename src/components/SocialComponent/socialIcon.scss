@import "../../styles/partials/colors";
@import "../../styles/partials/fonts";

.ui {
  &.image {
    cursor: pointer;
    width: 2.8rem;
  }

  &.ui.tiny.images {
    width: 2.5rem;
    margin-top: 1.875rem;
    height: auto;
    font-size: .85714286rem;
    text-align: center;
  }
}

.socialIconsHolder {
  text-align: center;
  margin: 1.25rem auto;

  &.socialIcon {
    text-align: center;
  }

  span {
    font-size: 0.5rem;
    display: inline-block;
  }
}

.groupIcon {
  width: 1.875rem;
  display: inline-block;
  margin-right: 0.938rem;
  margin-top: 0.625rem;
  cursor: pointer;
}

i.icon.facebook.official:before {
  content: "\f082";
  color: $facebook;
}

i.icon.twitter.square:before {
  content: "\f081";
  color: $twitter;
}

i.icon.google.plus.square:before {
  content: "\f0d4";
  color: $google
}

i.icon.linkedin:before {
  content: "\f08c";
  color: $linkedin
}