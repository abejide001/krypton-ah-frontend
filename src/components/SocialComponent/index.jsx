import React from 'react';
import { Icon } from 'semantic-ui-react';
import { socialMediaLoginIcons } from '../../constants';
import './socialIcon.scss';

function SocialIcons() {
  return (
    <div className="socialIconsHolder">
      {
        socialMediaLoginIcons.map(iconDetail => (
          <a href={iconDetail.url} key={iconDetail.key}>
            <div className="groupIcon">
              <Icon className="socialIcon" size="big" name={iconDetail.iconName} />
              <span>{iconDetail.text}</span>
            </div>
          </a>
        ))
      }
    </div>
  );
}

export default SocialIcons;
