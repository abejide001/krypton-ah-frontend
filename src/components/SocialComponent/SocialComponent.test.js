import React from 'react';
import { shallow } from 'enzyme';
import SocialComponent from '.';

describe('Modal Social', () => {
  it('it should matches the snapshot', () => {
    const wrapper = shallow(<SocialComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders four <Icon /> components', () => {
    const wrapper = shallow(<SocialComponent />);
    expect(wrapper.find('Icon')).toHaveLength(4);
  });
});
