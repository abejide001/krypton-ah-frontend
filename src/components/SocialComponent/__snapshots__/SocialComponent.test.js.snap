// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Social it should matches the snapshot 1`] = `
<div
  className="socialIconsHolder"
>
  <div
    className="groupIcon"
    key="1"
  >
    <Icon
      as="i"
      className="socialIcon"
      name="facebook official"
      size="big"
    />
    <span>
      Facebook
    </span>
  </div>
  <div
    className="groupIcon"
    key="2"
  >
    <Icon
      as="i"
      className="socialIcon"
      name="google plus square"
      size="big"
    />
    <span>
      Google
    </span>
  </div>
  <div
    className="groupIcon"
    key="3"
  >
    <Icon
      as="i"
      className="socialIcon"
      name="linkedin"
      size="big"
    />
    <span>
      Linkedin
    </span>
  </div>
  <div
    className="groupIcon"
    key="4"
  >
    <Icon
      as="i"
      className="socialIcon"
      name="twitter square"
      size="big"
    />
    <span>
      Twitter
    </span>
  </div>
</div>
`;
